x/stakeibc/keeper/send_host_zone_unbondings_test.go

```diff
@@ -0,0 +1,164 @@
+package keeper_test
+
+import (
+	// "fmt"
+
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	_ "github.com/stretchr/testify/suite"
+
+	recordtypes "github.com/Stride-Labs/stride/x/records/types"
+
+	// "github.com/Stride-Labs/stride/x/stakeibc/types"
+	stakeibc "github.com/Stride-Labs/stride/x/stakeibc/types"
+)
+
+type SendHostZoneUnbondingTestCase struct {
+	amt_to_unbond           uint64
+	epoch_unbonding_records []recordtypes.EpochUnbondingRecord
+	host_zone               stakeibc.HostZone
+	lightClientTime         uint64
+}
+
+func (s *KeeperTestSuite) SetupSendHostZoneUnbonding() SendHostZoneUnbondingTestCase {
+	host_val1_addr := "cosmos_VALIDATOR_1"
+	host_val2_addr := "cosmos_VALIDATOR_2"
+	delegationAddr := "cosmos_DELEGATION"
+	amt_to_unbond := uint64(1_000_000)
+	amt_val1 := uint64(2_000_000)
+	amt_val2 := uint64(1_000_000)
+	wgt_val1 := uint64(2)
+	wgt_val2 := uint64(1)
+	unbonding_time := uint64(1660348276)
+	light_client_time := unbonding_time + 1
+
+	//  define the host zone with stakedBal and validators with staked amounts
+	validators := []*stakeibc.Validator{
+		{
+			Address:       host_val1_addr,
+			DelegationAmt: amt_val1,
+			Weight:        wgt_val1,
+		},
+		{
+			Address:       host_val2_addr,
+			DelegationAmt: amt_val2,
+			Weight:        wgt_val2,
+		},
+	}
+
+	delegationAccount := stakeibc.ICAAccount{
+		Address: delegationAddr,
+		Target:  stakeibc.ICAAccountType_DELEGATION,
+	}
+
+	hostZone := stakeibc.HostZone{
+		ChainId:           "GAIA",
+		HostDenom:         "uatom",
+		Bech32Prefix:      "cosmos",
+		Validators:        validators,
+		DelegationAccount: &delegationAccount,
+	}
+
+	// list of epoch unbonding records
+	epochUnbondingRecords := []recordtypes.EpochUnbondingRecord{
+		recordtypes.EpochUnbondingRecord{
+			Id:                   1,
+			UnbondingEpochNumber: 1,
+			HostZoneUnbondings:   make(map[string]*recordtypes.HostZoneUnbonding),
+		},
+	}
+
+	// for each epoch unbonding record, add a host zone unbonding record and append the record
+	for _, epochUnbondingRecord := range epochUnbondingRecords {
+		// note: we're using the same hostZoneUnbonding for each epoch unbonding record in this test
+		epochUnbondingRecords[0].HostZoneUnbondings["GAIA"] = &recordtypes.HostZoneUnbonding{
```

Comment:
Sorry for the late review! This should be updated now that we're no longer using a map